src/flushMicroTasks.ts

```diff
@@ -2,7 +2,7 @@ import { setImmediate } from './helpers/timers';
 
 type Thenable<T> = { then: (callback: () => T) => unknown };
 
-export function flushMicroTasks<T>(): Thenable<T> {
+export function flushMicroTasks(): Thenable<void> {
   return {
     // using "thenable" instead of a Promise, because otherwise it breaks when
     // using "modern" fake timers
```

Comment:
This is a weird thenable because the `then` function doesn't return anything and is therefore not chainable. Also, the `setImmediate` timer starts running at the time when the promise chain is _constructed_, not when it _runs_.
```js
flushMicroTasks().then( flushMicroTasks ).then( flushMicroTasks )
```
should run there `setImmediate`s after each other, but instead it will crash, and even after fixing the crash, it will run all three timers in parallel.

The way how `await` works doesn't expose these bugs, but outside `await`, the helper won't work.

My intuition is that the really correct version will look differently 🙂

Reply:
My bad, I've missed that comment.

So when used this in instead of inline `Promise` it passed the test correctly. @jsnajdr would you be able to submit a PR with better implementation. You seem to now the Promise stuff very well, so help in this area would be very welcomed.

Reply:
Should the implementation be just:

```
import { setImmediate } from './helpers/timers';

export function flushMicroTasks() {
  return new Promise((resolve) => setImmediate(resolve));
}
```